resolve.go

```diff
@@ -9,59 +9,81 @@ import (
 )
 
 var ResolvableProtocols = []ma.Protocol{DnsaddrProtocol, Dns4Protocol, Dns6Protocol, DnsProtocol}
-var DefaultResolver = &Resolver{Backend: net.DefaultResolver}
+var DefaultResolver = &Resolver{def: net.DefaultResolver}
 
 const dnsaddrTXTPrefix = "dnsaddr="
 
-type Backend interface {
+// BasicResolver is a low level interface for DNS resolution
+type BasicResolver interface {
```

Comment:
Not sure if I have enough context here, but doesn't DNS allow us to lookup A, AAAA and TXT records in the same query? This interface suggests that looking up both would cause 2 queries (and potentially 2 roundtrips).

Reply:
Also, they have different usage, we don't do concurrent A/AAAA and TXT resolution in our usage patterns.

Reply:
To be fair, at some point may have some  overlap in our usage patterns. 
For example, a request for DNSLink might be accompanied by a lookup for a content routing hint (if we implement some form of https://github.com/ipfs/go-ipfs/issues/6516). 

That being said, I don't think we should do any premature optimizations.
When time comes, adding cache for raw DNS records (that respects their TTL) will be enough.